fe2-android/app/src/main/java/com/github/dedis/student20_pop/utility/network/IdGenerator.java

```diff
@@ -0,0 +1,106 @@
+package com.github.dedis.student20_pop.utility.network;
+
+import com.github.dedis.student20_pop.model.event.EventType;
+import com.github.dedis.student20_pop.utility.security.Hash;
+
+/** ID Generator class */
+public class IdGenerator {
+
+    static final String SUFFIX_ELECTION_QUESTION = "Question";
+
+    private IdGenerator() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    /**
+     * Generate the id for dataCreateLao and dataUpdateLao.
+     * https://github.com/dedis/student_21_pop/blob/master/protocol/query/method/message/data/dataCreateLao.json
+     * https://github.com/dedis/student_21_pop/blob/master/protocol/query/method/message/data/dataUpdateLao.json
+     *
+     * @param organizer ID of the organizer
+     * @param creation creation time of the LAO
+     * @param name original or updated name of the LAO
+     * @return the ID of CreateLao or UpdateLao computed as Hash(organizer||creation||name)
+     */
+    public static String generateLaoId(String organizer, long creation, String name) {
+        return Hash.hash(organizer, Long.toString(creation), name);
+    }
+
+    /**
+     * Generate the id for dataCreateMeeting.
+     * https://github.com/dedis/student_21_pop/blob/master/protocol/query/method/message/data/dataCreateMeeting.json
+     *
+     * @param laoId ID of the LAO
+     * @param creation creation time of the Meeting
+     * @param name original or updated name of the Meeting
+     * @return the ID of CreateMeeting computed as Hash('M'||lao_id||creation||name)
+     */
+    public static String generateCreateMeetingId(String laoId, long creation, String name) {
+        return Hash.hash(EventType.MEETING.getSuffix(), laoId, Long.toString(creation), name);
+    }
+
+    /**
+     * Generate the id for dataCreateRollCall.
+     * https://github.com/dedis/student_21_pop/blob/master/protocol/query/method/message/data/dataCreateRollCall.json
+     *
+     * @param laoId ID of the LAO
+     * @param creation creation time of RollCall
+     * @param name name of RollCall
+     * @return the ID of CreateRollCall computed as Hash('R'||lao_id||creation||name)
+     */
+    public static String generateCreateRollCallId(String laoId, long creation, String name) {
+        return Hash.hash(EventType.ROLL_CALL.getSuffix(), laoId, Long.toString(creation), name);
+    }
+
+    /**
+     * Generate the id for dataOpenRollCall.
+     * https://github.com/dedis/student_21_pop/blob/master/protocol/query/method/message/data/dataOpenRollCall.json
+     *
+     * @param laoId ID of the LAO
+     * @param opens id of RollCall to open
+     * @param openedAt open time of RollCall
+     * @return the ID of OpenRollCall computed as Hash('R'||lao_id||opens||opened_at)
+     */
+    public static String generateOpenRollCallId(String laoId, String opens, long openedAt) {
```

Comment:
Why not call the argument idRollCall (or something in that spirit) rather than opens?

Reply:
The idea was to follow the protocol specifications as close as possible in order to avoid confusion and mistakes